klippy/extras/led.py

```diff
@@ -60,10 +61,15 @@ def _check_transmit(self, print_time=None):
         if not self.need_transmit:
             return
         self.need_transmit = False
-        try:
-            self.update_func(self.led_state, print_time)
-        except self.printer.command_error as e:
-            logging.exception("led update transmit error")
+        def reactor_cb(eventtime):
+            try:
+                # Just avoid any race conditions
+                led_state = self.led_state
```

Comment:
Thanks.  I think the assignment `led_state = self.led_state` should occur in `_check_transmit()` - that is just before `def reactor_cb()`.  As, it is possible for a state change to occur even between `reactor.register_callback()` and the invocation of `reactor_cb()`.

-Kevin

Reply:
Good catch!